### PR TITLE
add referrers api 

### PR DESCRIPTION
**Pull request by wy65701436**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Review by stonezdj — APPROVED** [review]
LGTM

File: src/server/registry/referrers.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	"github.com/goharbor/harbor/src/pkg/artifact"
+	"github.com/goharbor/harbor/src/server/router"
+	"github.com/goharbor/harbor/src/server/v2.0/handler"
+)
+
+const ReferrersSchemaVersion = 2
+const ReferrersMediaType = "application/vnd.oci.image.index.v1+json"
+
+func newReferrersHandler() http.Handler {
+	return &referrersHandler{}
```
Comment (chlins) [inline src/server/registry/referrers.go:24]: Don't need to initialize the two fields(artifactManager/accessoryManager) in the structure?

**Review by chlins — APPROVED** [review]
lgtm

File: src/server/registry/referrers.go (hunk 1 above)
Comment (wy65701436) [reply src/server/registry/referrers.go]: updated

**Pull request merged**